### ISSUE TITLE
heroku_build local_checksum constantly changing

### ISSUE THREAD

**Issue by ryanking** [heroku/terraform-provider-heroku#289]
### Terraform Version

v0.12.x

### Heroku Provider Version

3.0.0

### Affected Resource(s)

- heroku_build

### Expected Behavior

Using `heroku_build`'s `path` attribute with a directory should only trigger a new build if the contents of the files in the directory change.

### Actual Behavior

We use terraform enterprise, which means that each time we run our code, the files will have different metadata (ex: modified time).  This means that the tarball generated by the provider will have a different checksum on each run.

### Proposal

I propose (and am willing to contribute) that when `heroku_build`'s `path` is set to a directory that the checksum be calculate before building the tarball and that we write the code such that metadata like modified time are not included in the checksum.

### References

- #269 (note some comments about using terraform cloud that fit with what we are seeing)

**Comment by davidji99** [comment]
Is only the modified time changing for the file metadata in Terraform Enterprise? If so, ignoring this metadata could be a great idea.

If there can be more metadata changes, perhaps a `provider` level attribute that allows users to specify what metadata to ignore when doing a checksum comparison could be a better route.

**Comment by mars** [comment]
👍 I agree that the checksum strategy should be improved. This original "checksum the tarball" technique was naive in that regard, and it clearly needs to be fixed for Terraform Cloud users.

**Comment by mraerino** [comment]
is there any way to work around this?

**Comment by mars** [comment]
A workaround would be to use `heroku_build`'s `source.url` instead of `source.path`:

Make GitHub releases for your app, and point Terraform at its archive URLs:
https://registry.terraform.io/providers/heroku/heroku/latest/docs/resources/build#github-urls

If not using GitHub, you can point `source.url` to any URL that returns the Heroku build-compatible source tarball:
https://registry.terraform.io/providers/heroku/heroku/latest/docs/resources/build#source-urls

**Comment by vood** [comment]
@mars I'm willing to provide some help here. This issue is quite annoying and the workarounds you propose are not that easy to implement as one can think for private repos (how can I set up heroku to pull private releases?).

here is the code structure that fails
```
/terraform/heroku/main.tf <-- this is where heroku_build is called
/src/heroku-app/heroku.yml <-- here is where my heroku app along with Dockerfile
```

**Comment by mars** [comment]
I believe I have a working solution to this problem, by relaxing the checksum for source directory to only cover file paths & contents. See PR #319

**Comment by mars** [comment]
Fixed in [release v4.6.0](https://github.com/heroku/terraform-provider-heroku/releases/tag/v4.6.0) 🥳
